components/dashboards-web-component/src/designer/components/PagesSidePane.jsx

```diff
@@ -170,6 +179,33 @@ class PagesSidePane extends Component {
         });
     }
 
+    /**
+     * Update dashboard page visibility.
+     *
+     * @param {string} pageId Page ID
+     * @param {boolean} hidden Is hidden
+     * @returns {Promise}
+     */
+    handlePageVisibility(pageId, hidden) {
+        return new Promise((resolve, reject) => {
+            const page = this.getPageById(pageId);
+            page.hidden = hidden;
+            this.props.updateDashboard()
+                .then(() => resolve())
+                .catch(() => {
+                    delete page.hidden;
```

Comment:
Instead of deleting, we should revert the change.
See [`handleLandingPageUpdate` function](https://github.com/wso2/carbon-dashboards/pull/985/files#diff-817114f3567404d06ed20e6d48cbbd02R126).

Reply:
Ack